static/js/util/profile_edit.js

```diff
@@ -282,9 +289,5 @@ export function saveProfileStep(validator: Validator|UIValidator, isLastStep: bo
   let clone = Object.assign({}, profile, {
     filled_out: profile.filled_out || isLastStep
   });
-  if (isLastStep) {
-    // user has also seen email consent message at this point
-    clone.email_optin = true;
-  }
```

Comment:
@aliceriot only issue with this logic is if profile is not filled_out then if user selects ``email_optin = false`` on the ``isLastStep == true`` last step the this info with not save, because we are making it ``clone.email_optin = true;`` always